spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/controller/StreamController.java

```diff
@@ -214,9 +216,14 @@ private void deployStream(StreamDefinition stream, Map<String, String> deploymen
 		Iterator<ModuleDefinition> iterator = stream.getDeploymentOrderIterator();
 		for (int i = 0; iterator.hasNext(); i++) {
 			ModuleDefinition module = iterator.next();
-			String type = (i == 0) ? "sink"
-					: (iterator.hasNext() ? "processor" : "source");
-			ModuleCoordinates coordinates = this.registry.findByNameAndType(module.getName(), type);
+			ModuleType type = (i == 0) ? ModuleType.sink
```

Comment:
I believe the idea is that on the SCS side there isn't strict checking of module types, but on the SCD side it will be more opinionated. 

@markfisher and I went back and forth on this one a few times. I can see the possibility of going with strings if it's possible to add a new module type without modifying SCD...